src/Spd.Manager.Licence/BizPortalUserManager.cs

```diff
@@ -110,4 +120,37 @@ public async Task<BizPortalUserListResponse> Handle(BizPortalUserListQuery query
             Users = userResps
         };
     }
+
+    public async Task<Unit> Handle(BizPortalUserUpdateLoginCommand cmd, CancellationToken ct)
+    {
+        await _portalUserRepository.ManageAsync(
+               new PortalUserUpdateLoginCmd(cmd.UserId),
+               ct);
+        return default;
+    }
+
+    public async Task<Unit> Handle(BizPortalUserDeleteCommand request, CancellationToken cancellationToken)
+    {
+        //check role max number rule
+        
+        //TODO: ask if OrgId should = BizId
+        var existingUsersResult = (PortalUserListResp) await _portalUserRepository.QueryAsync(
+            new PortalUserQry() { OrgId = request.BizId},
+            cancellationToken);
+        var toDeleteUser = existingUsersResult.Items.FirstOrDefault(u => u.Id == request.UserId);
+        var newUsers = existingUsersResult.Items.ToList();
+        if (toDeleteUser == null) return default;
+        newUsers.Remove(toDeleteUser);
+        var biz = (BizResult) await _bizRepository.QueryBizAsync(new BizsQry(request.BizId), cancellationToken);
+
+        //TODO: ask what is primaryUserNo and how to fill ContactRoleCode (or ContactAuthorizationTypeCode)
```

Comment:
There is a 'MaximumNumberOfAuthorizedContacts' and a 'MaximumNumberOfPrimaryAuthorizedContacts' that should be enforced for each business.

Reply:
where should exactly enforce it? there is a func at line 149 `CheckMaxRoleNumberRuleAsync`

Reply:
You could do it the same way it is done in Screening... check OrgUserManager.cs... it calls 'SharedManagerFuncs.CheckMaxRoleNumberRuleAsync'